examples/nanocoap_server/coap_handler.c

```diff
@@ -97,11 +97,16 @@ static ssize_t _riot_value_handler(coap_pkt_t *pkt, uint8_t *buf, size_t len, vo
     case COAP_PUT:
     case COAP_POST:
     {
-        /* convert the payload to an integer and update the internal value */
-        char payload[16] = { 0 };
-        memcpy(payload, (char*)pkt->payload, pkt->payload_len);
-        internal_value = strtol(payload, NULL, 10);
-        code = COAP_CODE_CHANGED;
+        if (pkt->payload_len <= 15) {
```

Comment:
Where does 15 come from?

Reply:
I haven taken it from /examples/gcoap/gcoap_cli.c line 202. Perhaps space should be allocated for the null character.

Reply:
why is the payload even copied ?

Reply:
to ensure `strtol()` is called with a null terminated string?

Reply:
`internal_value` is a uint8, wouldn't 3 bytes be enough?

Reply:
ah it's related to the 16 byte buffer below